pkg/client/client.go

```diff
@@ -278,7 +287,7 @@ func ApplyUnstructuredObjectsWithNewLabels(ctx context.Context, cl client.Client
 	for _, unstructuredObj := range unstructuredObjects {
 		log.Info("applying object", "object_namespace", unstructuredObj.GetNamespace(), "object_name", unstructuredObj.GetObjectKind().GroupVersionKind().Kind+"/"+unstructuredObj.GetName())
 		MergeLabels(unstructuredObj, newLabels)
-		_, err := applyClient.ApplyObject(ctx, unstructuredObj)
+		_, err := applyClient.ApplyObject(ctx, unstructuredObj, SaveConfiguration(false))
```

Comment:
Unfortunately for Unstructured objects it is still not working, so I'm keeping the saveConfiguration functionality disabled. I've tried multiple things:

- removing the `LastAppliedConfigurationAnnotationKey` annotation from both new object before creating it
- removing the`LastAppliedConfigurationAnnotationKey` annotation from the existing object before merging the annotations with the new object
- removing the special handling of `Unstructured` objects in `marshalObjectContent`:
````
if newRes, ok := newResource.(runtime.Unstructured); ok {
		return json.Marshal(newRes.UnstructuredContent())
}
````

but nothing seems to work. The `toolchain.dev.openshift.com/last-applied-configuration` annotation is still being  recursively embedded.

Reply:
I'm also wondering if we really need this functionality at all. We could

a. trigger the Update of the object anyway , that should do anything if there are no changes on the object. 
b. rely on `kubectl.kubernetes.io/last-applied-configuration` , how is this different from our custom annotation ?

Reply:
Good points @mfrancisc. Not sure why we started using our annotation instead. Maybe @MatousJobanek remembers why.

Reply:
it was an attempt to get closer to what kubectl does, to optimize the number of updates we do from our controllers and to have some control over what was updated and what not. 
But I really doubt that it works as we hoped that it could work, so I'm for dropping the annotation :+1: to simplify the logic. But let's do it in a separate PR.